### PR TITLE
add HTTP proxy support

### PR DESCRIPTION
**Pull request by mathiask88**
WIP / Test - Do not merge

### CODE REVIEW

**Comment by mathiask88** [comment]
This change should handle proxies. It is not tested and we also need a small change in `simple-get` because `simple-get` drops the `reqOpts` object for redirects.
[This](https://github.com/mafintosh/prebuild-install/pull/13/files#diff-dc64f35b02398a7d4cfe87e136074600R53) should also work, but then the proxy can see the traffic. It is a potential security risk, but I think in our case it is not because we do not deal with secure data. The secure way would be to send a `CONNECT` to the proxy and then use the replied connection, but this tunneling is not possible with `simple-get`.

**Comment by mathiask88** [comment]
Proxy authentication is not implemented yet but no hard work. And it would be cool if someone using a proxy could test this :)

/cc @ralphtheninja @mafintosh @piranna @juliangruber

**Comment by piranna** [comment]
Sorry, not proxy here :-)

**Comment by mathiask88** [comment]
I set up a squid proxy and tested a bit. 
HTTP -> Proxy -> HTTP works
HTTPS -> Proxy -> .. does not work because most proxies do not understand TLS
HTTP -> Proxy -> HTTPS does not work (501 response) -> Proxy can not handle TLS
So we need to go with HTTP tunneling because prebuilds are 99% hosted on HTTPS (like github)

**Comment by mathiask88** [comment]
I will add https://github.com/request/tunnel-agent as a dependency to add _https over http tunneling_ and update this PR

**Comment by lgeiger** [comment]
@mathiask88 That's excellent. Really looking forward to it!

**Comment by ralphtheninja** [comment]
@lgeiger Want to wait with this PR until we release a new version?

**Comment by mathiask88** [comment]
The new commit enables tunneling if a proxy is set. I left the first commit to see the difference because maybe you don't want to always tunnel. Then we need a mix between the first and the second commit. I tested this only with HTTPS over HTTP.

File: download.js [hunk 1]
```diff
@@ -41,7 +42,33 @@ function downloadPrebuild (opts, cb) {
...
+        var reqOpts = downloadUrl
+        var parsedDownloadUrl = url.parse(downloadUrl)
+        var parsedProxy = null
+        var isHttpsRequest = parsedDownloadUrl.protocol === 'https:'
+
+        if (opts['https-proxy'] && isHttpsRequest) {
+          parsedProxy = url.parse(opts['https-proxy'])
+
+          if (parsedProxy.protocol === 'https:') {
+            return cb('https-proxies currently not supported')
```
Comment (ralphtheninja) [inline download.js]:
```js
return cb(new Error('https-proxies currently not supported'))
```

Comment (lgeiger) [inline download.js]: That's already fixed in the latest commit 👍

**Comment by lgeiger** [comment]
> Want to wait with this PR until we release a new version?

 I'm happy to include this PR in the new version.

File: download.js (hunk 1 above)
Comment (ralphtheninja) [inline download.js]: Aaah dang. I was trigger happy :)

**Comment by ralphtheninja** [comment]
2.0.0

**Pull request merged**